dhis2/src/main/java/org/motechproject/dhis2/service/impl/SettingsServiceImpl.java

```diff
@@ -60,10 +62,28 @@ private synchronized void loadSettings() {
             LOGGER.debug("Loading DHIS2 settings...");
             Gson gson = new Gson();
             settings = gson.fromJson(jsonText, Settings.class);
+            if (settings.getServerURI() != null) {
+                removeTrailingSlash();
+            }
         } catch (Exception e) {
             String message = "There was an error loading json from the DHIS2 settings.";
             LOGGER.debug(message);
             throw new JsonIOException(message, e);
         }
     }
+
+    private void removeTrailingSlash() {
+        try {
+            URI uri = new URI(settings.getServerURI());
+            uri = uri.normalize();
+            String uriString = uri.toString();
+            if (uriString.endsWith("/")) {
+                settings.setServerURI(uriString.substring(0, uriString.length() - 1));
+            }
+        } catch (URISyntaxException e) {
```

Comment:
duplicated code, this method should throw this exception and method above should catch this